tests/layering/layering_test.go

```diff
@@ -0,0 +1,148 @@
+	// See https://docs.rs/ostree-ext/0.5.1/ostree_ext/container/struct.OstreeImageReference.html
+	ostreeUnverifiedRegistry = "ostree-unverified-registry"
+	imageRegistry            = "image-registry.openshift-image-registry.svc:5000"
+	// If this moves from /run, make sure files get cleaned up
+	authfilePath   = "/run/ostree/auth.json"
+	mcoNamespace   = "openshift-machine-config-operator"
+	imagePullSpec  = "registry.ci.openshift.org/rhcos-devel/rhel-coreos:%s"
+	helloWorldPath = "/usr/bin/hello-world"
+	imageURL       = ostreeUnverifiedRegistry + ":" + imageRegistry + "/" + mcoNamespace + "/" + imageStreamName
+)
+
+type Deployments struct {
+	Booted                  bool   `json:"booted"`
+	ContainerImageReference string `json:"container-image-reference"`
+}
+
+type Status struct {
+	Deployments []Deployments `json:"deployments"`
+}
+
+func TestBootInClusterImage(t *testing.T) {
+	cs := framework.NewClientSet("")
+	// Get a random node to run on. Note: We don't need to set the infra role
+	// because this test does not involve the MCO.
+	targetNode, err := getRandomNode(cs, "worker")
+	require.Nil(t, err)
+
+	t.Logf("targeting node %s", targetNode.Name)
+
+	// If the delete-build flag is used, delete the Build and ImageStream afterward.
+	if deleteBuild != nil && *deleteBuild == true {
+		defer func() {
+			t.Logf("Deleting the ImageStream")
+			require.Nil(t, cs.ImageStreams(mcoNamespace).Delete(context.TODO(), imageStreamName, metav1.DeleteOptions{}))
+			t.Logf("Deleting the Image Build")
+			require.Nil(t, cs.BuildV1Interface.Builds(mcoNamespace).Delete(context.TODO(), buildName, metav1.DeleteOptions{}))
+		}()
+	}
+
+	canDeleteMachine := false
+	defer func() {
+		// Only if the image is applied should we delete
+		if canDeleteMachine {
+			if deleteMachine != nil && *deleteMachine == true {
+				deleteMachineAndNode(t, cs, targetNode)
+			} else {
+				t.Logf("leaving node %s behind, you need to clean it up manually", targetNode.Name)
+			}
+		} else {
+			t.Logf("node %s will not be deleted since it was not touched", targetNode.Name)
+		}
+	}()
+
+	testCases := []struct {
+		name     string
+		testFunc func(*testing.T)
+	}{
+		{
+			name: "Derived Image Is Built",
+			testFunc: func(t *testing.T) {
+				if err := maybeBuildDerivedOSImage(t, cs); err != nil {
+					t.Fatal(err)
+				}
+			},
+		},
+		{
+			name: "Not In Derived Image",
+			testFunc: func(t *testing.T) {
+				assertNotInDerivedImage(t, cs, targetNode)
+			},
+		},
+		{
+			name: "Boots Into Derived Image",
+			testFunc: func(t *testing.T) {
+				nodeRebooted := false
+				deletePullSecret := putPullSecretOnNode(t, cs, targetNode)
+				defer func() {
+					// The pull secret should be cleared from the node upon reboot since
+					// it is placed into the /run directory. However, if the node does
+					// not reboot and we're not deleting it, we should clean it up.
```

Comment:
I got confused about whether it was the pull secret or the node, maybe
```suggestion
					// not reboot or get deleted, we should clean it up.
```